sequencer/src/block2.rs

```diff
@@ -58,13 +66,168 @@ impl BlockPayload {
         payload.extend(tx_bodies);
         Self { payload }
     }
+
+    // Return the range `r` such that the `index`th tx bytes are at `self.payload[r]`.
+    fn get_tx_range(&self, index: TxIndex) -> Option<Range<usize>> {
+        let tx_bodies_offset = self.tx_bodies_offset()?;
+
+        // tx_table[i] is end index for the ith tx,
+        // so the range for the ith tx is tx_table[i-1..i].
+        // But tx_table starts at index 1 in the payload
+        // because index 0 in the payload is the length of the tx_table.
+        // So the range for the ith tx in the payload is actually
+        // [index..index+1].
+        let start = if index == 0 {
+            0
+        } else {
+            self.get_value_usize(index)?
+        }
+        .checked_add(tx_bodies_offset)?;
+
+        let end = self
+            .get_value_usize(index.checked_add(1)?)?
+            .checked_add(tx_bodies_offset)?;
+
+        Some(start..end)
+    }
+
+    // Viewing `self.payload` bytes as a vec of words of type `TxIndex`,
+    // return the `index`th word.
+    fn get_value(&self, index: TxIndex) -> Option<TxIndex> {
+        // TODO idiomatic usize <-> TxIndex conversion.
+        let start = usize::try_from(index)
+            .unwrap()
```

Comment:
@sveitser regarding https://github.com/EspressoSystems/espresso-sequencer/pull/776#issuecomment-1827542376:

> > * Some (such as idiomatic usize <-> u32 conversion) could be addressed immediately if we want, ...
> 
> @ggutoski I think if this is something we'll do anyway it might be good to do it now because it looks like it would make the code easier to read.

Agree. My comment is a cry for help / solicitation of opinions on how best to proceed. Options I see:

1. Fail the build if `usize` is too small to hold a `u32` / `TxIndex`. Good because we don't need error handling. (Though we still might need some error handling for `usize` -> `u32` conversion.)
2. Panic if conversion fails.
3. Use `Result` / `Option` for failed conversion.

I prefer something clean with no panic and no error handling cruft such as item 1. But in general what's best practice here?